libs/vertexai/tests/integration_tests/test_rank.py

```diff
@@ -0,0 +1,73 @@
+from unittest.mock import Mock, create_autospec, patch
+
+import pytest
+from google.api_core import exceptions as core_exceptions
+from google.cloud import discoveryengine_v1alpha
+from langchain.retrievers.contextual_compression import ContextualCompressionRetriever
+from langchain_core.documents import Document
+from langchain_core.vectorstores import VectorStore, VectorStoreRetriever
+from pytest import approx
+
+from langchain_google_vertexai.ranker.rank import VertexAIRanker
+
+
+# Fixtures for common setup
+@pytest.fixture
+def mock_rank_service_client():
+    mock_client = Mock(spec=discoveryengine_v1alpha.RankServiceClient)
+    mock_client.rank.return_value = discoveryengine_v1alpha.RankResponse(
+        records=[
+            discoveryengine_v1alpha.RankingRecord(
+                id="1", content="Document 1", title="Title 1", score=0.9
+            ),
+            discoveryengine_v1alpha.RankingRecord(
+                id="2", content="Document 2", title="Title 2", score=0.8
+            ),
+        ]
+    )
+    return mock_client
+
+
+@pytest.fixture
+def ranker(mock_rank_service_client):
+    return VertexAIRanker(
+        project_id="test-project",
+        location_id="test-location",
+        ranking_config="test-config",
+        title_field="source",
+        client=mock_rank_service_client,
+    )
+
+
+class MockVectorStoreRetriever(VectorStoreRetriever):
+    def _get_relevant_documents(self, query):
+        return [
+            Document(page_content="Document 1", metadata={"source": "Title 1"}),
+            Document(page_content="Document 2", metadata={"source": "Title 2"}),
+        ]
+
+
+@pytest.fixture
+def mock_vector_store_retriever():
+    mock_store = create_autospec(VectorStore, instance=True)
+    return MockVectorStoreRetriever(vectorstore=mock_store)
+
+
+@patch(
+    "langchain_google_vertexai.ranker.rank.discoveryengine_v1alpha.RankServiceClient"
+)
+def test_compress_documents(
+    mock_rank_service_client_class, ranker, mock_vector_store_retriever
+):
+    compression_retriever = ContextualCompressionRetriever(
```

Comment:
it doesn't look like an integration test to me since it mocks everything

can we maybe write an actual integration tests that triggers actual reranker?